### DISCUSSION TITLE
ng-mocks & ViewChild

### DISCUSSION THREAD

**Discussion by LukasMachetanz** [help-me-mom/ng-mocks#1947] [Q&A] [answered]
Hey! 

I am curious how ng-mocks ensures the correct behavior in combination with `ViewChild`; e.g. using it like `ViewChild(MyCompoment)` having `MyComponent` mocked with `MockComponent`. This [article](https://indepth.dev/posts/1184/angular-unit-testing-viewchild) describes the problem with the usual naive approach very well. Does ng-mocks use the described solution or is an alternative approach used? 

Thanks. Greets,
Lukas

**Comment by satanTime** [comment, marked as answer]
Hi there,

I think we could extend this documentation: https://ng-mocks.sudo.eu/guides/view-child

Feel free to let me know what is missing there.

**Reply by LukasMachetanz** [reply]
Hey, the documentation is perfectly fine for using it. I am more interested in the actual implementation and the approach how this works behind the scenes. :)

**Reply by satanTime** [reply]
There is nothing special. `ng-mocks` adds to `providers` info that a mock should be used when its real class is requested, the rest is done by angular.
https://github.com/ike18t/ng-mocks/blob/v13.0.4/libs/ng-mocks/src/lib/mock/decorate-declaration.ts#L48
